File: app/src/main/java/knf/kuma/database/dao/RecordsDAO.java
package knf.kuma.database.dao;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import knf.kuma.pojos.RecordObject;

@Dao
public interface RecordsDAO {
    @Query("SELECT * FROM recordobject ORDER BY date DESC")
    LiveData<List<RecordObject>> getAll();

    @Query("SELECT * FROM recordobject ORDER BY date DESC")
    List<RecordObject> getAllRaw();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void add(RecordObject object);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addAll(List<RecordObject> list);

    @Delete
    void delete(RecordObject object);

    @Query("DELETE FROM recordobject")
    void clear();
}

File: app/src/main/java/knf/kuma/database/dao/RecordsDAO.kt
package knf.kuma.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Query
import knf.kuma.pojos.RecordObject

@Dao
interface RecordsDAO {
    @get:Query("SELECT count(*) FROM recordobject")
    val count: Int
    @get:Query("SELECT * FROM recordobject ORDER BY date DESC")
    val allRaw: MutableList<RecordObject>

    @get:Query("SELECT aid FROM recordobject GROUP BY aid")
    val allAid: List<String>

    @Query("SELECT * FROM recordobject WHERE aid=:aid")
    fun findByAid(aid: String): List<RecordObject>

    @Query("DELETE FROM recordobject WHERE aid = :aid")
    fun deleteAllAid(aid: String): Int

    @Delete
    fun delete(recordObject: RecordObject)
}
